File: python/oop_basic/hawk.py
from animal import Animal

# INHERITANCE: Hawk gets alive/eat()/sleep()/aging() from Animal untouched
# (no override here, so Hawk.sleep() falls back to Animal's version).
class Hawk(Animal):
    wing = None

    def __init__(self, age, wing):
        self._age = age
        self._wing = wing

File: java/oop_basic/Hawk.java
// INHERITANCE: Hawk gets alive/eat()/sleep()/aging() from Animal untouched
// (no override here, so Hawk.sleep() falls back to Animal's version).
public class Hawk extends Animal {
    public int wing;

    public Hawk(int age, int wing) {
        super(age);
        this.wing = wing;
    }
}
